### PR TITLE
Add waveguide mode converter readme, schematic diagram, design patterns, and script

### PR DESCRIPTION
**Pull request by ianwilliamson**
First step in tackling #6. Specifically, this PR adds a readme to document the parameters and add a schematic for the waveguide mode converter. This is based on the results from our [paper](https://arxiv.org/abs/2201.12965).

### CODE REVIEW

**Comment by stevengj** [comment]
Great, thanks!  Can you also add a CSV file for data of the the optimized structure, or will that be a separate PR?

**Comment by ianwilliamson** [comment]
> Great, thanks! Can you also add a CSV file for data of the the optimized structure, or will that be a separate PR?

Yes, will follow up with additional PRs for (1) code to run the simulations and (2) the design patterns.

File: waveguide_mode_converter/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Waveguide Mode Converter
+
+This is a waveguide mode converter test problem. The design objective is to maximize the conversion of power from the
+fundamental waveguide mode of the input waveguide to the second-order mode of the output waveguide. The geometry is two-dimensional.
```
Comment (stevengj) [inline waveguide_mode_converter/README.md:4]: Be explicit about the objective function.

**Comment by stevengj** [comment]
Thanks for adding the CSV files; are these documented in the README?

**Comment by ianwilliamson** [comment]
> Thanks for adding the CSV files; are these documented in the README?

Not yet. Will take a stab at documenting them and adding a script for running the simulations.

For FOM / optimization objective, I am thinking that we can report worst case transmission over the bandwidth of interest. I will have the script print this out.

File: waveguide_mode_converter/README.md (hunk 1 above)
Comment (ianwilliamson) [reply waveguide_mode_converter/README.md]: Done.

**Comment by ianwilliamson** [comment]
> Will take a stab at documenting them and adding a script for running the simulations.
> 
> For FOM / optimization objective, I am thinking that we can report worst case transmission over the bandwidth of interest. I will have the script print this out.

Documentation has been added. I also add the script for reporting the performance metrics. 

A follow up step will be to have the script run the ruler on the designs, but currently this is not possible as the ruler.py script is not setup to be imported as a Python module. It also seems like there is a copy of ruler.py under the meta lens module.

**Pull request merged**